query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs

```diff
@@ -495,12 +497,22 @@ pub async fn query_raw<'conn>(
 
                 match operation {
                     MongoOperation::Find(filter, options) => {
-                        let cursor = coll.find_with_session(filter, options, session).await?;
+                        let unwrapped_filter = filter.clone().unwrap_or_default();
+                        let project = Document::default();
+                        let query_string_builder = Find::new(&unwrapped_filter, &project, coll.name());
+                        let cursor = observing(&query_string_builder, || {
+                            coll.find_with_session(filter, options, session)
+                        })
+                        .await?;
 
                         raw::cursor_to_json(cursor, session).await?
                     }
                     MongoOperation::Aggregate(pipeline, options) => {
-                        let cursor = coll.aggregate_with_session(pipeline, options, session).await?;
```

Comment:
I `grep`ed the codebase and these two places seem to be the only ones missing an `observing` call. Let me know if you find anything else

Reply:
Couldn't find any more either 👍